oss_src/lambda/worker_pool.hpp

```diff
@@ -91,43 +95,114 @@ std::unique_ptr<worker_process<ProxyType>> spawn_worker(std::vector<std::string>
 
   // Step 2: create cppipc client and connect it to the launched process 
   size_t retry = 0;
-  size_t max_retry = 3;
+
+  timer conn_timer;
+  conn_timer.start();
+
   std::unique_ptr<cppipc::comm_client> new_client;
-  while (retry < max_retry) {
-    // Make sure the server process is still around
+
+  auto check_process_exists = [&]() -> bool {
+    // Make sure the server process is still around.
     if (!new_process->exists()) {
-      logstream(LOG_ERROR) << "Lambda worker process terminated unexpectedly" << std::endl;
-      break;
+      int ret_code = new_process->get_return_code();
+
+      std::ostringstream ss;
```

Comment:
It isn't locking in a multithreaded environment, so you get interleaving of the output if you have a lot of << parts.   A single << works fine though.